### PR TITLE
🐛 Add ManagedBy to assets

### PR DESCRIPTION
**Pull request by czunker**
This field is needed for garbage collection.

Fixes https://github.com/mondoohq/mondoo-operator/issues/893

### CODE REVIEW

**Comment by czunker** [comment]
During my tests with a local minikube, I already saw that `ManagedBy` is now persisted for the assets. I need to wait some time, to also see whether the garbage collection also cleans them up.

File: policy/scan/local_scanner.go [hunk 1]
```diff
@@ -265,10 +249,11 @@ func (s *LocalScanner) distributeJob(job *Job, ctx context.Context, upstream *up
 		for _, a := range runtime.Provider.Connection.GetInventory().GetSpec().GetAssets() {
 			a.AddMondooLabels(resolvedAsset)
 			a.AddAnnotations(resolvedAsset.GetAnnotations())
+			a.ManagedBy = resolvedAsset.ManagedBy
```
Comment (czunker) [inline policy/scan/local_scanner.go:252]: This line is the actual fix. The rest of the PR is refactoring to allow adding a test case.

**Comment by czunker** [comment]
In combination with https://github.com/mondoohq/mondoo-operator/pull/903, this fixes https://github.com/mondoohq/mondoo-operator/issues/893

Screenshot from 2023-10-26 13-53-22

After the cleanup:
Screenshot from 2023-10-27 07-13-50

**Review by imilchev — APPROVED** [review]
:shipit:

**Pull request merged**